pgscatalog.match/src/pgscatalog/match/cli/intersect_cli.py

```diff
@@ -93,24 +93,35 @@ def run_intersect():
     o_tmp_t = []
     target_heap = []
     for path in args.target:
+        # assume inputs always have two extensions (bim.zst / pvar.zst)
+        path = pathlib.Path(path)
+        basename = path.name
+        stem = pathlib.Path(basename.split(".")[0])
+
         logger.info("Reading TARGET variants: {}".format(path))
         pvar = read_var_general(
             path, chrom=None
         )  # essential not to filter target (messes up common line indexing)
 
-        loc_afreq = path.replace(".pvar.zst", ".afreq.gz")
+        loc_afreq = path.parent / stem.with_suffix(".afreq.gz")
         afreq = read_var_general(
             loc_afreq, chrom=None
         )  # essential not to filter target (messes up common line indexing)
 
-        loc_vmiss = path.replace(".pvar.zst", ".vmiss.gz")
+        loc_vmiss = path.parent / stem.with_suffix(".vmiss.gz")
         vmiss = read_var_general(
             loc_vmiss, chrom=None
         )  # essential not to filter target (messes up common line indexing)
 
         for v, freq, miss in zip(pvar, afreq, vmiss):
-            if all([v["ID"], freq["ID"], miss["#ID"]]) is False:
-                raise ValueError("TARGET variant files are not sorted")
+            try:
+                is_sorted = all([v["ID"], freq["ID"], miss["#ID"]])  # pvar input
+            except KeyError:
+                is_sorted = all([v["ID"], freq["ID"], miss["ID"]])  # bim input
+            finally:
+                if not is_sorted:
+                    raise ValueError("TARGET variant files are not sorted")
+
```

Comment:
Why not just the if? There's no difference between the bim and pvar lines.

Reply:
when I started debugging there was a `KeyError` because the wrong files were getting read, and I forgot to tidy this up 😅 thanks!